feat: add API types for secrets sync controller (#1364)
kubernetes-sigs/secrets-store-csi-driver: secretsstoresynccontroller/README.md

> Secrets Store Sync Controller for Kubernetes secrets - Synchronizes secrets as Kubernetes secrets using the same interface as the [Secrets Store CSI Driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver) and is a Kubernetes sub-project [SIG Auth](https://github.com/kubernetes/community/tree/master/sig-auth).

Reviewer: Kubernetes SIG Auth sub-project

## Getting Started

> 1. The controller can be deployed as a deployment.  

Reviewer: ```suggestion
1. The controller is deployed as a deployment.  
```

1. The provider must be deployed as a container in the same pod as the controller. 
1. The controller watches for changes to the SecretStoreSync and SecretProviderClass objects and synchronizes the secrets defined in the SecretStoreSync as Kubernetes secrets.
1. The arguments passed to the controller are: 
    1. provider-volume - The volume name used to communicate with the provider. 

>     1. token-request-audience - Token requests audience for the controller. This is similar to the configuration required for the CSI Driver. Refer to [doc](https://kubernetes-csi.github.io/docs/token-requests.html) for more information.

Reviewer: I wouldn't point to the CSI docs for reference here because `token-request-audience` here is a list of strings while the CSI token requests is part of CSI spec and has a different schema.
